pegen/pegen.h

```diff
@@ -109,6 +111,18 @@ void *CONSTRUCTOR(Parser *p, ...);
 #define EXTRA start_lineno, start_col_offset, end_lineno, end_col_offset, p->arena
 
 PyObject *new_identifier(Parser *, char *);
+
+#define PROPAGATE_ERROR(p) \
+    longjmp(p->error_env, 1); \
+    Py_UNREACHABLE();
```

Comment:
It's super dangerous to have two statements in one macro. You'll have to surround it with `do { ... } while(0)` like we do everywhere.

Alternatively, why does this have to be a macro? It could just call a function that calls `longjmp()`. That might save a little object code.